packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -176,12 +182,13 @@ class ChangeNotifier implements Listenable {
   @override
   void removeListener(VoidCallback listener) {
     assert(_debugAssertNotDisposed());
-    for (final _ListenerEntry entry in _listeners!) {
-      if (entry.listener == listener) {
-        entry.unlink();
-        return;
-      }
-    }
+    final List<_ListenerEntry>? links = _index![listener];
+    if (links == null)
+      return;
+    assert(links.isNotEmpty);
+    links.removeAt(0).unlink();
```

Comment:
I'm pretty sure this won't be O(1) when you have to remove the same listener twice, which is probably an unusual case.

This is trading a bunch of space for time. It'd be really great to have a benchmark to help us understand just how much extra space we're using to optimize this.

Reply:
RemoveAt has to shift the items down, which is O(N).

Reply:
I see it now. Thank you.

Would you like me to replace the list with a ListQueue?

Reply:
Done. Also, the default, slower LinkedHashMap has been replaced with HashMap. The linked-ness was not needed here.